modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/dataset_display/views/table/TableHeadRow.js

```diff
@@ -117,25 +192,12 @@ function TableHeadRow({
 	showActionItems,
 	sorting,
 	updateSorting,
+	visibleFields,
 }) {
-	const getColumns = (fields) => {
-		const expandableColumns = fields.reduce(
-			(expandable, field) => expandable || Boolean(field.expand),
-			false
-		);
-
-		return fields.map((field, i) => {
-			return (
-				<TableHeadCell
-					{...field}
-					expandableColumns={expandableColumns}
-					key={field.sortingKey || i}
-					sorting={sorting}
-					updateSorting={updateSorting}
-				/>
-			);
-		});
-	};
+	const expandableColumns = visibleFields.reduce(
+		(expandable, field) => expandable || Boolean(field.expand),
+		false
+	);
```

Comment:
I know the number of fields is unlikely to be big, but the `reduce` is going to iterate over every field even though it could short-circuit.

If I read this correctly, the following would be a short-circuiting equivalent, which is arguably more readable too:

```js
const expandableColumns = visibleFields.some(field => field.expand);
```